app/main/posts/views/post-view-data.directive.js

```diff
@@ -88,6 +88,41 @@ function PostViewDataController(
      */
     var newPostsAfter = moment().utc();
     $scope.savingPost = {saving: false};
+
+    $rootScope.$on('event:edit:post:data:mode:saveSuccess', function () {
+        function removePostFromList() {
+            for (var i = 0; i < $scope.posts.length; i++) {
+                if ($scope.posts[i].id === $scope.selectedPostId) {
+                    let nextInLine = $scope.posts[i + 1];
+                    $scope.posts.splice(i, 1);
+                    groupPosts($scope.posts);
+                    $scope.selectedPost.post = nextInLine;
+                    $scope.selectedPostId = nextInLine.id;
+                }
+            }
+        }
+
+        if ($scope.hasFilters()) {
+
+            let query = PostFilters.getQueryParams($scope.filters);
+
+            let postQuery = _.extend({}, query, {
+                limit: $scope.itemsPerPage
+            });
+
+            postQuery.offset = ($scope.currentPage - 1) * $scope.itemsPerPage;
+
+            PostEndpoint.query(postQuery).$promise.then(function (postsResponse) {
+                for (let i = 0; i < postsResponse.results.length; i++) {
+                    if (postsResponse.results[i].id === $scope.selectedPostId) {
+                        return;
+                    }
+                }
+                removePostFromList();
```

Comment:
I don't understand what this is doing? it looks like you're *always* removing it from the list?